src/hooks/useInitialWindowDimensions/index.js

```diff
@@ -0,0 +1,58 @@
+// eslint-disable-next-line no-restricted-imports
+import {useState, useEffect} from 'react';
+import {Dimensions} from 'react-native';
+import {initialWindowMetrics} from "react-native-safe-area-context";
+
+/**
+ * A convenience hook that provides initial size (width and height).
+ * An initial height allows to know the real height of window,
+ * while the standard useWindowDimensions hook return the height minus Virtual keyboard height
+ */
+export default function () {
+    const [dimensions, setDimensions] = useState(() => {
+        const window = Dimensions.get('window');
+        const screen = Dimensions.get('screen');
+
+        return {
+            screenHeight: screen.height,
+            screenWidth: screen.width,
+            initialHeight: window.height,
+            initialWidth: window.width,
+        };
+    });
+
+    useEffect(() => {
+        const onDimensionChange = (newDimensions) => {
+            const {window, screen} = newDimensions;
+
+            setDimensions((oldState) => {
+                if (screen.width !== oldState.screenWidth || screen.height !== oldState.screenHeight || window.height > oldState.initialHeight) {
+                    return {
+                        initialHeight: window.height,
+                        initialWidth: window.width,
+                        screenHeight: screen.height,
+                        screenWidth: screen.width,
+                    };
+                }
+
+                return oldState;
+            });
+        };
+
+        const dimensionsEventListener = Dimensions.addEventListener('change', onDimensionChange);
+
+        return () => {
+            if (!dimensionsEventListener) {
+                return;
+            }
+            dimensionsEventListener.remove();
+        };
+    }, []);
+
+    const bottomInset = initialWindowMetrics?.insets?.bottom ?? 0;
+
+    return {
+        initialWidth: dimensions.initialWidth,
+        initialHeight: dimensions.initialHeight - bottomInset,
```

Comment:
I haven't used insets from ScreenWrapper, because it shows not insets that I want.
It seems it does not take into account the bottom "Home bar" and `insets.bottom` returns `0`. So I've googled how to fix this issue, and found out the solution that it is need to use insets from `initialWindowMetrics` and it returns correct bottom inset.